src/main/java/DukeFunctions/Event.java

```diff
@@ -0,0 +1,41 @@
+package DukeFunctions;
+
+import Exceptions.MissingInputException;
+
+public class Event extends Todo {
+
+    protected String from;
+    protected String to;
+    String deliverable;
+
+    public Event(String inputContents) throws MissingInputException {
+        super(inputContents);
+        String[] parts = inputContents.split("/from|/to");
+
+        String deliverable = parts[0].trim();
+        if (parts.length < 3) {
+
+            throw new MissingInputException();
+        } else {
+            from = (parts.length > 1) ? parts[1].trim() : "";
+            to = (parts.length > 2) ? parts[2].trim() : "";
+        }
+        this.deliverable = deliverable;
+        this.type = "E";
```

Comment:
Try to refactor all magic literals into constants to improve readability. See topics on code quality on the course website.